programs/monaco_protocol/src/instructions/math.rs

```diff
@@ -32,15 +32,31 @@ pub fn stake_precision_is_within_range(stake: u64, decimal_limit: u8) -> bool {
 }
 
 pub fn calculate_commission(commission_rate: f64, profit: i128) -> u64 {
+    if profit <= 0 || commission_rate == 0.0 {
+        return 0;
+    }
+
     let commission_rate_decimal = Decimal::from_f64(commission_rate).unwrap();
     Decimal::from(profit)
-        .max(Decimal::ZERO)
         .mul(commission_rate_decimal)
         .div(Decimal::ONE_HUNDRED)
         .to_u64()
         .unwrap()
 }
 
+pub fn calculate_post_commission_remainder(commission_rate: f64, profit: i128) -> u64 {
+    if profit <= 0 || commission_rate == 0.0 {
```

Comment:
I don't think we can also return 0 when commission rate is zero here, since this function returns the remainder. When `commission_rate == 0` then the result of the function will be `profit`.